__tests__/application/index.test.js

```diff
@@ -24,28 +25,6 @@ describe('app', () => {
       .end(() => {})
   })
 
-  it('should not .writeHead when !socket.writable', done => {
```

Comment:
why was this test removed?

Reply:
a very strange test that has a stuck connection

Reply:
@jonathanong, do you have thoughts on the safety of removing this test? I'm concerned that since it is passing currently either `node:test` has an issue with this or the `// hackish, but the response should occur in a single tick` might be failing to "hack it" anymore. In either case, it seems like it is here verify core doesn't crash/create errors when socket isn't writeable, so might be problematic to remove.

Maybe pass this along but open a critical bug to re-add it?

Reply:
i'm okay with removing it since it is a hack. also, i think the API for the socket might've changed since this was originally written.

Reply:
I think the main thing we're testing here is that we don't try to write again if the headers are already sent